module/documents/actor/actor.mjs

```diff
@@ -581,27 +581,21 @@ export default class Actor5e extends SystemDocumentMixin(Actor) {
     const progression = { slot: 0, pact: 0 };
     const types = {};
 
-    // NPCs don't get spell levels from classes
-    if ( this.type === "npc" ) {
-      progression.slot = this.system.details.spellLevel ?? 0;
-      types.leveled = 1;
-    }
+    // Grab all classes with spellcasting
+    const classes = this.items.filter(cls => {
+      if ( cls.type !== "class" ) return false;
+      const type = cls.spellcasting.type;
+      if ( !type ) return false;
+      types[type] ??= 0;
+      types[type] += 1;
+      return true;
+    });
 
-    else {
-      // Grab all classes with spellcasting
-      const classes = this.items.filter(cls => {
-        if ( cls.type !== "class" ) return false;
-        const type = cls.spellcasting.type;
-        if ( !type ) return false;
-        types[type] ??= 0;
-        types[type] += 1;
-        return true;
-      });
+    for ( const cls of classes ) this.constructor.computeClassProgression(
+      progression, cls, { actor: this, count: types[cls.spellcasting.type] }
+    );
 
-      for ( const cls of classes ) this.constructor.computeClassProgression(
-        progression, cls, { actor: this, count: types[cls.spellcasting.type] }
-      );
-    }
+    if ( this.type === "npc" ) progression.slot += this.system.details.spellLevel ?? 0;
```

Comment:
Maybe this should act as an override rather than an addition?

Reply:
Hmm, taking a look back at the rules I'd probably vote the other way around, that any classes with spellcasting completely overwrite the `spellLevel` value. While the DMG "Monsters with Classes" rules don't mention existing spellcasting levels, Tasha's "Spellcaster" sidekick class says it replaces and existing spellcasting on the stat block. If we have class levels take priority over this level then things would work according to the rules by simply adding a class to the NPC, rather than having to explicitly remove the original `spellLevel` value first.

Reply:
Yeah, I'd agree with that